packages/dashboard-v2/src/components/forms/AccountSettingsForm.js

```diff
@@ -0,0 +1,111 @@
+import * as Yup from "yup";
+import PropTypes from "prop-types";
+import { Formik, Form } from "formik";
+
+import { Button } from "../Button";
+import { TextField } from "../Form/TextField";
+import accountsService from "../../services/accountsService";
+
+const isPopulated = (value) => value?.length > 0;
+
+const emailUpdateSchema = Yup.object().shape({
+  email: Yup.string().email("Please provide a valid email address"),
+  confirmEmail: Yup.string()
+    .oneOf([Yup.ref("email"), null], "Emails must match")
+    .when("email", {
+      is: isPopulated,
+      then: (schema) => schema.required("Please confirm new email address"),
+    }),
+  password: Yup.string().min(1, "Password can't be blank"),
+  confirmPassword: Yup.string()
+    .oneOf([Yup.ref("password"), null], "Passwords must match")
+    .when("password", {
+      is: isPopulated,
+      then: (schema) => schema.required("Please confirm new password"),
+    }),
```

Comment:
When updating the user's profile, we're fine with them not supplying a password - their password will stay the same. I'm not sure I'm reading this right but if I am, we are requiring a password change here.

Also, it should be fine for the user to not have an email.

Reply:
I created new issue with your comment to follow up.